metricflow/dataflow/builder/dataflow_plan_builder.py

```diff
@@ -157,6 +158,7 @@ def _build_metrics_output_node(
         queried_linkable_specs: LinkableSpecSet,
         where_constraint: Optional[SpecWhereClauseConstraint] = None,
         time_range_constraint: Optional[TimeRangeConstraint] = None,
+        combine_metrics_with_inner_join: bool = False,
```

Comment:
Can we add a docstring to explain the usage of this boolean?

Alternatively, we could remove the boolean in favor of a MetricType, something like `output_metric_type: Optional[MetricType] = None` and then set the join type based on a check for `output_metric_type is MetricType.DERIVED`.